thirdeye-persistence/src/main/java/ai/startree/thirdeye/datalayer/bao/NamespaceConfigurationManagerImpl.java

```diff
@@ -0,0 +1,246 @@
+
+  @Inject
+  public NamespaceConfigurationManagerImpl(final NamespaceConfigurationDao dao,
+      final TimeConfiguration timeConfiguration) {
+    this.dao = dao;
+    this.timeConfiguration = timeConfiguration;
+  }
+
+  public @NonNull NamespaceConfigurationDTO getNamespaceConfiguration(final String namespace) {
+    final NamespaceConfigurationDTO existingNamespaceConfig = fetchExistingNamespaceConfiguration(
+        namespace);
+
+    // namespace config exists, update if components are empty
+    if (existingNamespaceConfig != null) {
+      if (existingNamespaceConfig.getTimeConfiguration() == null) {
+        existingNamespaceConfig.setTimeConfiguration(
+            getDefaultTimeConfigurationFromServerConfig());
+        final Long namespaceConfigurationId = save(existingNamespaceConfig);
+        checkState(namespaceConfigurationId != null,
+            "Failed to update namespace configuration for namespace %s",
+            existingNamespaceConfig.namespace());
+      }
+      return existingNamespaceConfig;
+    }
+
+    return createNewNamespaceConfiguration(namespace);
+  }
+
+  public @NonNull NamespaceConfigurationDTO updateNamespaceConfiguration(
+      NamespaceConfigurationDTO updatedNamespaceConfiguration) {
+    String namespace = updatedNamespaceConfiguration.namespace();
+    final NamespaceConfigurationDTO existingNamespaceConfig = fetchExistingNamespaceConfiguration(
+        namespace);
+    checkState(existingNamespaceConfig != null,
+        "Trying to update non-existent namespace configuration for namespace %s",
+        namespace);
+
+    final Long namespaceConfigurationId = save(updatedNamespaceConfiguration);
+    checkState(namespaceConfigurationId != null,
+        "Failed to update namespace configuration for namespace %s",
+        namespace);
+
+    return updatedNamespaceConfiguration;
+  }
+
+  public @NonNull NamespaceConfigurationDTO resetNamespaceConfiguration(final String namespace) {
+    final NamespaceConfigurationDTO existingNamespaceConfig = fetchExistingNamespaceConfiguration(
+        namespace);
+
+    // namespace config exists, update values to default
+    if (existingNamespaceConfig != null) {
+      existingNamespaceConfig.setTimeConfiguration(
+          getDefaultTimeConfigurationFromServerConfig());
+      final Long namespaceConfigurationId = save(existingNamespaceConfig);
+      checkState(namespaceConfigurationId != null,
+          "Failed to rollback namespace configuration for namespace %s",
+          existingNamespaceConfig.namespace());
+      return existingNamespaceConfig;
+    }
+
+    return createNewNamespaceConfiguration(namespace);
+  }
+
+  private NamespaceConfigurationDTO fetchExistingNamespaceConfiguration(String namespace) {
+    final DaoFilter daoFilter = new DaoFilter().setPredicate(Predicate.EQ(
+        "namespace", namespace));
+    final List<NamespaceConfigurationDTO> results = filter(daoFilter);
+    if (results != null && !results.isEmpty()) {
```

Comment:
let's ensure the list only has a single element before .get(0).
if it has multiple elements, the db is in an inconsistent state (or somehow the query above is not valid anymore) and we should not return any configuration

Reply:
done. it should throw exception in an invalid state